homeassistant/components/shelly/__init__.py

```diff
@@ -320,6 +326,39 @@ async def _async_setup_rpc_entry(hass: HomeAssistant, entry: ShellyConfigEntry)
         await hass.config_entries.async_forward_entry_setups(
             entry, runtime_data.platforms
         )
+
+        if runtime_data.rpc_supports_scripts and is_firmware_update_available(
+            runtime_data.rpc.model
+        ):
+            firmware = AwesomeVersion(device.shelly["ver"])
+            issue_id = BLE_SCANNER_FIRMWARE_UNSUPPORTED_ISSUE_ID.format(
+                unique=entry.unique_id
+            )
+            if (
+                firmware < BLE_SCANNER_MIN_FIRMWARE
+                and entry.options.get(CONF_BLE_SCANNER_MODE) == BLEScannerMode.ACTIVE
+            ):
+                ir.async_create_issue(
+                    hass,
+                    DOMAIN,
+                    issue_id,
+                    is_fixable=True,
+                    is_persistent=True,
+                    severity=ir.IssueSeverity.WARNING,
+                    translation_key="ble_scanner_firmware_unsupported",
+                    translation_placeholders={
+                        "device_name": device.name,
+                        "ip_address": device.ip_address,
+                        "firmware": firmware,
+                    },
+                    data={"entry_id": entry.entry_id},
+                )
+            else:
+                ir.async_delete_issue(
+                    hass,
+                    DOMAIN,
+                    issue_id,
+                )
```

Comment:
Sorry for being unclear. I was thinking of putting this whole block in its own function.

Reply:
Oh, didn't get it, I'll change that.